source/d3d9/d3d9_device.cpp

```diff
@@ -456,7 +456,24 @@ HRESULT STDMETHODCALLTYPE Direct3DDevice9::EndScene()
 }
 HRESULT STDMETHODCALLTYPE Direct3DDevice9::Clear(DWORD Count, const D3DRECT *pRects, DWORD Flags, D3DCOLOR Color, float Z, DWORD Stencil)
 {
-	return _orig->Clear(Count, pRects, Flags, Color, Z, Stencil);
+	com_ptr<IDirect3DSurface9> depthstencil;
+	_orig->GetDepthStencilSurface(&depthstencil);
```

Comment:
This call can be moved inside the first `if` below, so that it's not executed when only clearing the color buffer.